flexmeasures/data/schemas/scheduling/storage.py

```diff
@@ -54,36 +54,62 @@ class StorageFlexModelSchema(Schema):
     You can use StorageScheduler.deserialize_flex_config to get that filled in.
     """
 
-    soc_at_start = fields.Float(required=True, data_key="soc-at-start")
+    soc_at_start = QuantityField(
+        required=True,
+        to_unit="MWh",
+        default_src_unit="dimensionless",  # placeholder, overridden in __init__
+        return_magnitude=True,
+        data_key="soc-at-start",
+    )
 
-    soc_min = fields.Float(validate=validate.Range(min=0), data_key="soc-min")
-    soc_max = fields.Float(data_key="soc-max")
+    soc_min = QuantityField(
+        validate=validate.Range(
+            min=0
+        ),  # change to min=ur.Quantity("0 MWh") in case return_magnitude=False
+        to_unit="MWh",
+        default_src_unit="dimensionless",  # placeholder, overridden in __init__
+        return_magnitude=True,
+        data_key="soc-min",
+    )
+    soc_max = QuantityField(
+        to_unit="MWh",
+        default_src_unit="dimensionless",  # placeholder, overridden in __init__
+        return_magnitude=True,
+        data_key="soc-max",
+    )
 
-    power_capacity_in_mw = QuantityOrSensor(
+    power_capacity_in_mw = TimeSeriesOrQuantityOrSensor(
         "MW", required=False, data_key="power-capacity"
     )
 
-    consumption_capacity = QuantityOrSensor(
+    consumption_capacity = TimeSeriesOrQuantityOrSensor(
         "MW", data_key="consumption-capacity", required=False
     )
-    production_capacity = QuantityOrSensor(
+    production_capacity = TimeSeriesOrQuantityOrSensor(
         "MW", data_key="production-capacity", required=False
     )
 
-    # Timezone placeholder for the soc_maxima, soc_minima and soc_targets fields are overridden in __init__
-    soc_maxima = TimeSeriesOrSensor(
-        unit="MWh", timezone="placeholder", data_key="soc-maxima"
+    # Timezone placeholders for the soc_maxima, soc_minima and soc_targets fields are overridden in __init__
+    soc_maxima = TimeSeriesOrQuantityOrSensor(
+        to_unit="MWh",
+        default_src_unit="dimensionless",  # placeholder, overridden in __init__
+        timezone="placeholder",
+        data_key="soc-maxima",
     )
 
-    soc_minima = TimeSeriesOrSensor(
-        unit="MWh",
+    soc_minima = TimeSeriesOrQuantityOrSensor(
+        to_unit="MWh",
+        default_src_unit="dimensionless",  # placeholder, overridden in __init__
         timezone="placeholder",
         data_key="soc-minima",
         value_validator=validate.Range(min=0),
     )
 
-    soc_targets = TimeSeriesOrSensor(
-        unit="MWh", timezone="placeholder", data_key="soc-targets"
+    soc_targets = TimeSeriesOrQuantityOrSensor(
+        to_unit="MWh",
+        default_src_unit="dimensionless",  # placeholder, overridden in __init__
+        timezone="placeholder",
+        data_key="soc-targets",
     )
 
     soc_unit = fields.Str(
```

Comment:
Any *sensor*? Or do you mean if any numeric value is passed directly in any SoC related *field*?

Reply:
Sorry, I meant *field*. Now that we are allowing defining all SOC fields with sensor we don't need soc-unit anymore. If any of the fields requires it (time series or scalar values without units), raise a validation exception.

Reply:
I like what you are suggesting, i.e. to raise in case numeric values are given without an soc-unit. However, it would unfortunately be a breaking change. Currently, if not passed, the soc-unit is guessed from the unit of the power sensor that's being scheduled. That happens [here](https://github.com/FlexMeasures/flexmeasures/blob/ec6fd7c829749e22894319c5043af969f118dbae/flexmeasures/data/models/planning/storage.py#L467-L472) (since https://github.com/FlexMeasures/flexmeasures/pull/572).

If a breaking change is indeed needed, I would like to do that in a separate PR.

Reply:
Oh, good to know. Now I notice: the default value for `required` is `False`. Could we dataset it explicitly?